src/test/java/com/wikia/webdriver/testcases/auth/SignupTests.java

```diff
@@ -0,0 +1,284 @@
+package com.wikia.webdriver.testcases.auth;
+
+import com.wikia.webdriver.common.contentpatterns.MercurySubpages;
+import com.wikia.webdriver.common.contentpatterns.MercuryWikis;
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.annotations.InBrowser;
+import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.helpers.*;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.mercury.components.TopBar;
+import com.wikia.webdriver.elements.mercury.pages.ArticlePage;
+import com.wikia.webdriver.elements.mercury.pages.discussions.PostsListPage;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.global_navitagtion.NavigationBar;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.register.DetachedRegisterPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.register.RegisterPage;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+import java.time.LocalDate;
+
+import static com.wikia.webdriver.common.core.Assertion.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+
+@Execute(onWikia = MercuryWikis.MERCURY_AUTOMATION_TESTING)
+public class SignupTests extends NewTestTemplate {
+
+  private static final String USERNAME_TAKEN_MSG = "Username is taken";
+  private static final String PASSWORD_MATCHING_USERNAME_MSG = "Password and username cannot match";
+  private static final String GENERIC_ERROR_MSG =
+    "We cannot complete your registration at this time";
+
+  private static final String DESKTOP = "auth-signup-desktop";
+  private static final String MOBILE = "auth-signup-mobile";
+  private static final String PASS_PATTERN = "pass_%s";
+  private static final String USERNAME_PATTERN = "QA%s";
+
+  private UserWithEmailPool userPool = new UserWithEmailPool();
+  private UserWithEmail userWithEmail = userPool.getEmailOnlyUser1();
+  private User existingUser = User.LOGIN_USER;
+  private static final LocalDate BIRTH_DATE = LocalDate.of(1993, 3, 19);
+
+
+
+  @Test(groups = DESKTOP)
+  public void newUserCanSignUpDesktop() {
+    performSuccessfulSignUpOnDesktopAs(createNewUser());
```

Comment:
Can you `performUnsuccessfulSignUpOnDesktopAs(user)`? If not maybe `performSignUpOnDesktopAs(user)`.

Reply:
Okay, you can `performUnsuccessfulSignUpOnDesktopAs`. =)
Still, I think you don't need to add `successful`.